buildGoCache.nix

```diff
@@ -66,8 +66,14 @@ buildGoModule {
     xargs go build <${importPackagesFile}
     mkdir -p $out/nix-support
     cat > $out/nix-support/setup-hook <<EOF
-      cp --reflink=auto -r $out/go-cache $TMPDIR/go-cache
-      chmod -R +w $TMPDIR/go-cache
+      mkdir -p ''$TMPDIR || true;
+      if [ -d "''$TMPDIR" ]; then
+        echo copying $out/go-cache ''$TMPDIR/go-cache
+        cp --reflink=auto -r $out/go-cache ''$TMPDIR/go-cache
+        chmod -R +w ''$TMPDIR/go-cache
```

Comment:
```suggestion
      mkdir -p "$TMPDIR" || true;
      if [ -d "$TMPDIR" ]; then
        echo "copying ${placeholder "out"}/go-cache "$TMPDIR/go-cache"
        cp --reflink=auto -r "${placeholder "out"}/go-cache" "$TMPDIR/go-cache"
        chmod -R +w "$TMPDIR/go-cache"
```

Reply:
```suggestion
      mkdir -p "$TMPDIR" || true
      if [ -d "$TMPDIR" ]; then
        echo "copying ${placeholder "out"}/go-cache" "$TMPDIR/go-cache"
        cp --reflink=auto -r "${placeholder "out"}/go-cache" "$TMPDIR/go-cache"
        chmod -R +w "$TMPDIR/go-cache"
```

Reply:
Does `"$var"` escape the `$` inside `''   ''` too like `''$var`?

Reply:
With `'EOF'` the `$out` variable won't work so have to get the out dir through string interpolation

Reply:
But when I debug it via builtins.trace I get `trace: /1rz4g4znpzjwh1xymhjpm42vipw92pr73vdgl6xs1hycac8kf2n9` for `placeholder "out"`, doesn't look anything like a nix store path?

Reply:
It's replaced by nix, see https://github.com/search?q=repo%3ANixOS%2Fnix%201rz4g4znpzjwh1xymhjpm42vipw92pr73vdgl6xs1hycac8kf2n9&type=code

Reply:
> Edit: Ok so nix will later replace the generated placeholder string (like above) with the correct out path. But what is the correct "$out" it will be replaced with? $out of whichever derivation is being built?

Yes we want to hardcode the out of the current derivation so once the hook is run it'll be copied